### ISSUE TITLE
`DELETE /users` endpoint for front-end UAT tests to use

### ISSUE THREAD

**Issue by bpkennedy** [pbbg/api.pbbg.com#12]
The front-end UAT's need to be able to "start fresh", database-wise, before each test.  An easy way to facilitate this is to create a /deleteAll endpoint which essentially truncates the db tables of their data (while keeping the latest schema and mandatory seed data).

For security reasons, we wouldn't want this endpoint to function in production environments.

**Comment by foohonpie** [comment]
Let's put it on the regular endpoints, that way we can call them selectively and keep closer to REST.

So we'll need:

- [ ] `DELETE /users` API definition

- [ ] phpunit tests

- [ ] code changes

We can make a separate ticket for gating it, as it may tie into some admin stuff that also needs doing.

**Comment by foohonpie** [comment]
@bpkennedy are there currently UAT needs to delete other records, or just users?

I'd like to get this on all endpoints eventually, but having it on just the 1 may be a good first issue for somebody to pick up.

**Comment by bpkennedy** [comment]
Currently we are only creating new Users, no other models yet.  Once Create A Game is ready then we'll be doing those, too.

I'd say if there's a clean way to generically have the delete all on each resource then that would work well.

**Comment by bpkennedy** [comment]
This is accomplished with #21 !  As stated earlier, there should be a followup story on restricting use of this endpoint to Admin users only.
